### ISSUE TITLE
Creating a new foreign key with onDelete results in "undefined" ON DELETE in query.

### ISSUE THREAD

**Issue by cthuluhoop123** [knex/knex#4250]
# Environment

Knex version: 0.21.16
Database + version: sqlite3
OS: Windows 10
Node: 14.15.4

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

Using knex migration to create a foreign key column works until you use `.onDelete('SET NULL');` where the raw query ends up with `...ON DELETE undefined`, causing the migration query to fail.

This only happens if the migration that creates the foreign key column is added AFTER an initial create schema migration (where the column is not initially created).

2. Error message

```
Error: CREATE TABLE `tickets` 
(
    `id` integer not null primary key autoincrement,
    `attendee_id` integer,
    `seat_id` integer,
    foreign key(`attendee_id`) references `attendees`(`id`) on delete SET NULL, 
    FOREIGN KEY (`seat_id`)  REFERENCES `mapped_seats` (`id`) ON DELETE undefined
) 
- SQLITE_ERROR: near "undefined": syntax error
```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

To reproduce, create an initial schema migration:
```js

exports.up = function (knex) {
    return knex.schema
        .createTable('table', table => {
            table
                .increments('id')
                .primary();
        })
        .createTable('table2', table => {
            table
                .increments('id')
                .primary();
        });
};

exports.down = function (knex) {
    return knex.schema
        .dropTableIfExists('table')
        .dropTableIfExists('table2');
};
```
Run the migration `knex migrate:latest`

Then create the second migration which adds a foreign key:

```js

exports.up = function (knex) {
    return knex.schema.table('table', table => {
        table
            .integer('table2_id')
            .references('id')
            .inTable('table2')
            .index()
            .onDelete('SET NULL');
    });
};

exports.down = function (knex) {
    return knex.schema.table('table', table => {
        table.dropColumn('table2_id');
    });
};

```

Resulting error/query: 
```
Error: CREATE TABLE `table` (`id` integer not null primary key autoincrement, `table2_id` integer,  FOREIGN KEY (`table2_id`)  REFERENCES `table2` (`id`) ON DELETE undefined) - SQLITE_ERROR: near "undefined": syntax error
```

**Comment by kibertoad** [comment]
Should be fixed in  0.21.17, would appreciate if you could confirm.

**Comment by cthuluhoop123** [comment]
Thanks, confirmed to be fixed.
